Testing: Add page about software testing with CrateDB (Java and Python) (#80)
crate/crate-clients-tools: docs/integrate/testing.md

(testing)=
# Software Testing with CrateDB

> This documentation section lists Java- and Python-based test frameworks

Reviewer: I think it's simpler to just say:
```suggestion
Java and Python based test frameworks...
```

Author: Thanks. I amended the patch, using your suggestions.

and libraries that support software integration testing with CrateDB.

(python-pytest)=
## Python pytest

> The popular [pytest] framework makes it easy to write small tests, yet
> scales to support complex functional testing for applications and libraries.

Reviewer: ```suggestion
The popular [pytest] framework makes it easy to write small tests, but
it also supports complex functional testing for applications and libraries.
```

Author: Used their original slogan, but I like yours much better. Thanks.

The [pytest-crate] package manages CrateDB instances for running integration
tests against them. It is based on [cr8](#cr8), and provides the `crate`,

> `crate_execute`, and `crate_cursor` pytest fixtures on top.

Reviewer: sorry, this `on top` is a bit confusing, on top of what? or do you mean crate_execute and crate_cursor on top of crate?

Author: Rephrasing to use:

> It is based on [cr8](#cr8) for the heavy lifting, and additionally provides
the `crate`, `crate_execute`, and `crate_cursor` pytest fixtures for
developer convenience.

- [Using "pytest-crate" with CrateDB and pytest]

(cr8)=
(python-unittest)=
## Python unittest

> [cr8], a collection of tools for CrateDB developers, provides primitives to

Reviewer: ```suggestion
[cr8], a collection of tools for CrateDB developers, provides primitive functionality, to
```

Author: I didn't want to say "primitive functionality", but rather "primitive element" instead. Do you have a different suggestion?

-- https://www.techtarget.com/whatis/definition/primitive

Author: I've amended the patch once more, and used "primitive elements" literally now. Maybe this causes less hiccups for the uninitiated?

Reviewer: it's minor detail, no need to change. Just, for me, it would read better as `basic functionality`, as this: `primitive elements` suit better in a programming language context.

Author: I see your point. In this case, the page about software testing probably has an audience of software developers. For me, it is difficult to designate something as `basic functionality`, when effectively it is `advanced functionality` already, wrapped into a primitive interface.

Now, that we talked about our personal dilemmas in wording extensively, let's merge the PR ;].
